recipes/glfw/all/conanfile.py

```diff
@@ -29,6 +29,9 @@ def _configure_cmake(self):
             self._cmake.definitions["GLFW_BUILD_DOCS"] = False
             if self.settings.compiler == "Visual Studio":
                 self._cmake.definitions["USE_MSVC_RUNTIME_LIBRARY_DLL"] = "MD" in self.settings.compiler.runtime
+            if self.options.vulkanStatic:
+                self._cmake.definitions["GLFW_VULKAN_STATIC"] = True
```

Comment:
Isn't the better solution to add [vulkan-loader](https://github.com/conan-io/conan-center-index/pull/2581) to the requires and do:
```python
self._cmake.definitions["GLFW_VULKAN_STATIC"] = not self.options["vulkan-loader"].shared
```

Reply:
But what if you don't want to link vulkan at all and load it dynamically, e.g. via [volk](https://github.com/conan-io/conan-center-index/tree/master/recipes/volk)?

Reply:
Good point.
So there are 3 possibilities:
1. No vulkan-loader in the requires, load it using a loader
2. Link to vulkan library dynamically
3. Link to vulkan library statically

Is this correct?

Would the following work?
```python
self._cmake.definitions["GLFW_VULKAN_STATIC"] = "vulkan-loader" in self.options and not self.options["vulkan-loader"].shared
```

Reply:
Technically there is another possibility: Don't link to vulkan at all and don't load it and use OpenGL ;-)

Which `vulkan-loader` package exactly would you use? Does it contain MoltenVK for Darwin and the LunarG vulkan libraries for Linux/Windows, are there no license issues for that?

For me, the suggested `vulkanStatic` option works fine by linking the Vulkan loader library that was manually installed (without conan) from the respective SDK.

Reply:
I meant vulkan-loader from khronos: see https://github.com/conan-io/conan-center-index/pull/2581

cci's aim is to provide a *complete experience*, so it would be nice to not have to manually install libraries, so we want to model the dependency.

I guess we will need to add moltenvk as a package and create a *meta package* for vulkan support.
CC'ing @SpaceIm for more opinions

Reply:
Right, it would be nice having Vulkan in conan. I think there is another linking variant: Create a shared glfw library (`-o glfw:shared=True`) which links to vulkan statically, while the application itself would only link to glfw, but not to any vulkan library. Is that possible with your suggestion? This would mean that glfw needs `vulkan-loader` as an _optional_ dependency (since not all users need vulkan at all), because the application using glfw would _not_ have a direct `vulkan-loader` dependency in that case.

Reply:
Didn't dig to much, but in my opinion:
- if `GLFW_VULKAN_STATIC` is TRUE, glfw should link to static `vulkan-loader`
- it is forbidden to link static vulkan into glfw shared: https://github.com/glfw/glfw/blob/96f9f5c4b970140ab5380fb9886ba787c8698937/CMakeLists.txt#L51, so conan recipe should check this in `configure` method.
- Last time I looked at `vulkan-loader` project, static build was removed except for Apple like OS: https://github.com/KhronosGroup/Vulkan-Loader/blob/399f7fd741f76f73832bbd11a7366acd06060862/loader/CMakeLists.txt#L185. It means that `GLFW_VULKAN_STATIC` only makes sense on Apple like OS.

Reply:
Okay, so should we wait until the `vulkan-loader` CCI recipe is ready and then do it as [proposed](https://github.com/conan-io/conan-center-index/pull/2927#discussion_r489580926) by @madebr, i.e. don't use a `vulkanStatic` option at all?